onnxruntime/test/fuzzing/src/OnnxPrediction.cpp

```diff
@@ -45,101 +62,83 @@ session{nullptr}
 
 // Destructor
 //
-OnnxPrediction::~OnnxPrediction()
-{
-  if (ptrSession.get() == &session)
-  {
-      // Ensure the session is not deleted
-      // by the unique_ptr. Because it will be deleting the stack
-      //
-      ptrSession.release();
+OnnxPrediction::~OnnxPrediction() {
+  if (ptrSession.get() == &session) {
+    // Ensure the session is not deleted
+    // by the unique_ptr. Because it will be deleting the stack
+    //
+    ptrSession.release();
   }
 }
```

Comment:
This seems somewhat egregious. I know it was existing code, but can we clean up the ownership semantics here? 

Whoever owns the session should have it in a unique_ptr and let that free it. Whoever is using the session but does not own it should just have a reference to the session. I don't really know what 'deleting the stack' means either. :)

Reply:
The session_ptr is nullptr if the OnnxPrediction object is created from the model file rather than the ModelProto. It seems the code is making sure that ptr_session actually points to the session object. Either session_ptr is nullptr or there is a session object created and needs to be "destroyed". In the OnnxPrediction::~OnnxPrediction function we can safely check and delete (reset) ptr_session.

---
In reply to: [521682737](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521682737) [](ancestors = 521682737)